rust/oak/src/lib.rs

```diff
@@ -107,22 +96,36 @@ thread_local! {
 /// }
 /// ```
 pub fn set_node<T: Node + 'static>() {
-    // TODO: Detect multiple invocations.
     NODE.with(|node| {
-        *node.borrow_mut() = Box::new(T::new());
+        match *node.borrow_mut() {
+            Some(ref mut _node) => {
+                writeln!(logging_channel(), "attempt to set_node() when already set!").unwrap();
+                panic!("attempt to set_node when already set");
+            }
+            None => {
+                writeln!(logging_channel(), "setting current node instance").unwrap();
+            }
+        }
+        *node.borrow_mut() = Some(Box::new(T::new()));
     });
 }
 
 #[no_mangle]
 pub extern "C" fn oak_handle_grpc_call() {
-    NODE.with(|node| {
-        let mut grpc_method_channel = Channel::new(GRPC_METHOD_NAME_CHANNEL_HANDLE);
-        let mut grpc_method_name = String::new();
-        grpc_method_channel
-            .read_to_string(&mut grpc_method_name)
-            .unwrap();
-        let mut grpc_channel = Channel::new(GRPC_CHANNEL_HANDLE);
-        node.borrow_mut()
-            .invoke(&grpc_method_name, &mut grpc_channel);
+    NODE.with(|node| match *node.borrow_mut() {
+        Some(ref mut node) => {
+            let mut grpc_method_channel = Channel::new(GRPC_METHOD_NAME_CHANNEL_HANDLE);
+            let mut grpc_method_name = String::new();
+            grpc_method_channel
+                .read_to_string(&mut grpc_method_name)
+                .unwrap();
+            let mut grpc_channel = Channel::new(GRPC_CHANNEL_HANDLE);
+            node.invoke(&grpc_method_name, &mut grpc_channel);
+        }
+        None => {
+            let mut logging_channel = logging_channel();
+            writeln!(logging_channel, "gRPC call with no loaded Node").unwrap();
```

Comment:
In other places `logging_channel()` was inlined, perhaps keep it consistent.

Reply:
Done.